### ISSUE TITLE
URLSession and AHC transports double-encode path parameters

### ISSUE THREAD

**Issue by sliemeobn** [apple/swift-openapi-generator#251]
When using String path parameters, the generated client (correctly) URL-encodes the values to be included in the path.
However, the server handler just passes the the "raw" string value in the `input.path` structure.

Eg:
using the generated client:
```swift
        let response = try await client.createRole(.init(
            path: .init(roleId: "A role"),
            ...
        ))
```

server code
```swift
    func createRole(_ input: Operations.createRole.Input) async throws -> 
      //input.path.roleId is "A%20role"
}
```

I would expect this to be symmetric and arrive decoded in the handler.

**Comment by czechboy0** [comment]
Thanks for filing, yes that looks like a bug.

**Comment by czechboy0** [comment]
I tried to reproduce in a unit test, but seems to be behaving correctly: https://github.com/apple/swift-openapi-runtime/pull/49

Possibly this could be caused by the client using 0.2 and server 0.1? I could see that leaving the client as escaped, but not being unescaped on the server.

**Comment by sliemeobn** [comment]
found it, it is actually a bug in the [URLSessionTransport's initializer](https://github.com/apple/swift-openapi-urlsession/blob/294b1e958bb3a82f967c94c130ad86fcee0af6c0/Sources/OpenAPIURLSession/URLSessionTransport.swift#L160-L168) of `URLRequest`

The URLComponents.url adds another round of percent encoding, basically sending an double-encoded path parameter to the server.

**Comment by czechboy0** [comment]
Ok both the URLSession and AHC transports have been released with this fix in 0.2.2.
